cmake/linker/ld/target.cmake

```diff
@@ -52,6 +52,52 @@ macro(configure_linker_script linker_script_gen linker_pass_define)
   )
 endmacro()
 
+# Force symbols to be entered in the output file as undefined symbols
+function(toolchain_ld_force_undefined_symbols)
+  foreach(symbol ${ARGN})
+    zephyr_link_libraries(-u${symbol})
+  endforeach()
+endfunction()
+
+# Link a target to given libraries with toolchain-specific argument order
+#
+# Usage:
+#   toolchain_ld_link_elf(
+#     TARGET_ELF             <target_elf>
+#     LIBRARIES_PRE_SCRIPT   [libraries_pre_script]
+#     LINKER_SCRIPT          <linker_script>
+#     LIBRARIES_POST_SCRIPT  [libraries_post_script]
+#     DEPENDENCIES           [dependencies]
+#   )
+function(toolchain_ld_link_elf)
+  cmake_parse_arguments(
+    TOOLCHAIN_LD_LINK_ELF                                     # prefix of output variables
+    ""                                                        # list of names of the boolean arguments
+    "TARGET_ELF;LINKER_SCRIPT"                                # list of names of scalar arguments
+    "LIBRARIES_PRE_SCRIPT;LIBRARIES_POST_SCRIPT;DEPENDENCIES" # list of names of list arguments
+    ${ARGN}                                                   # input args to parse
+  )
+
+  target_link_libraries(
+    ${TOOLCHAIN_LD_LINK_ELF_TARGET_ELF}
+    ${TOOLCHAIN_LD_LINK_ELF_LIBRARIES_PRE_SCRIPT}
+    ${TOPT}
+    ${TOOLCHAIN_LD_LINK_ELF_LINKER_SCRIPT}
+    ${TOOLCHAIN_LD_LINK_ELF_LIBRARIES_POST_SCRIPT}
+
+    ${LINKERFLAGPREFIX},-Map=${PROJECT_BINARY_DIR}/${KERNEL_MAP_NAME}
```

Comment:
I can't imagine that specifying a map file is order-dependent, so it should also be placed in an abstraction and moved out of here.

Reply:
Alright. Rather than another abstraction, another named argument to this function `toolchain_ld_link_elf` seems appropriate.